qcfractal/tests/test_services.py

```diff
@@ -120,6 +120,94 @@ def test_service_torsiondrive_duplicates(torsiondrive_fixture):
     base_run, duplicate_run = procedures
     assert base_run.optimization_history == duplicate_run.optimization_history
 
+def test_service_torsiondrive_optional(torsiondrive_fixture):
+    """"Tests torsiondrive with optional keywords"""
+
+    spin_up_test, client = torsiondrive_fixture
+
+    ## test optional dihedral_ranges feature
+    ret = spin_up_test(keywords={"grid_spacing": [30], "dihedral_ranges": [[-150, -60]]})
+    # Get a TorsionDriveORM result and check data
+    result = client.query_procedures(id=ret.ids)[0]
+    assert result.status == "COMPLETE"
+    # check final energies
+    final_energies = result.get_final_energies()
+    # the dihedral range should be limited to -150, -90, -60
+    assert set(final_energies.keys()) == {(-150,), (-120,), (-90,), (-60,)}
+    assert pytest.approx(0.0005683235570009067, abs=1e-6) == final_energies[(-150,)]
+    assert pytest.approx(0.0002170694130912583, abs=1e-6) == final_energies[(-120,)]
+    assert pytest.approx(0.0001565537585121726, abs=1e-6) == final_energies[(-90,)]
+    assert pytest.approx(0.0007991274441437338, abs=1e-6) == final_energies[(-60,)]
+    # check final molecules
+    final_molecules = result.get_final_molecules()
+    assert set(final_molecules.keys()) == {(-150,), (-120,), (-90,), (-60,)}
+    assert all(hasattr(m, "symbols") for m in final_molecules.values())
+
+    ## test optional energy_decrease_thresh feature
```

Comment:
Can you split each of these up into separate tests rather than a single lump?

Reply:
Sure I have separated this part into 3 test functions.